clients/roscpp/include/ros/timer_manager.h

```diff
@@ -428,11 +428,36 @@ void TimerManager<T, D, E>::setPeriod(int32_t handle, const D& period)
   {
     return;
   }
+  
+  if(reset)
+  {
+    info->next_expected = T::now() + period;
+  }
+  
+  // If called in a callback, last_real has not been updated yet so (now - last_real) > period
+  // In this case, let next_expected be updated only in updateNext
```

Comment:
Tested. I have a small node I am using to just run a Timer's callback, call setPeriod, and report the time that elapses between each callback. Should I commit it? If so, where?